src/test/java/com/teragrep/aer_01/EventContextConsumerTest.java

```diff
@@ -52,96 +52,99 @@
             EventContext eventContext = eventContextFactory.create();
             eventContextConsumer.accept(eventContext);
+        }
 
-            long depth1 = 0L;
-            final double records = 10;
-            for (int i = 1; i < records; i++) { // records - 1 loops
-                eventContext = eventContextFactory.create();
-                eventContextConsumer.accept(eventContext);
+        long latency = Instant.now().getEpochSecond();
 
-                if (i == 4) {
-                    depth1 = eventContext.getLastEnqueuedEventProperties().getOffset() - eventContext.getEventData().getOffset();
-                }
-            }
+        // 5 records for each partition
+        Gauge<Long> gauge1 = metricRegistry.gauge(name(EventContextConsumer.class, "latency-seconds", "1"));
+        Gauge<Long> gauge2 = metricRegistry.gauge(name(EventContextConsumer.class, "latency-seconds", "2"));
 
-            long depth2 = eventContext.getLastEnqueuedEventProperties().getOffset() - eventContext.getEventData().getOffset();
-            Gauge<Long> gauge1 = metricRegistry.gauge(name(EventContextConsumer.class, "depth-bytes", "1"));
-            Gauge<Long> gauge2 = metricRegistry.gauge(name(EventContextConsumer.class, "depth-bytes", "2"));
-
-            Assertions.assertEquals(depth1, 99L); // offsets are defined in the factory
-            Assertions.assertEquals(depth2, 99L);
-            Assertions.assertEquals(depth1, gauge1.getValue());
-            Assertions.assertEquals(depth2, gauge2.getValue());
-        }
+        // hard to test the exact correct latency
+        Assertions.assertTrue(gauge1.getValue() >= latency);
+        Assertions.assertTrue(gauge2.getValue() >= latency);
     }
 
     @Test
-    public void testEstimatedDataDepthMetric() throws Exception {
-        final EventContextFactory eventContextFactory = new CheckpointlessEventContextFactory();
-        final Sourceable configSource = new PropertySource();
-        final int prometheusPort = new MetricsConfig(configSource).prometheusPort;
-        final MetricRegistry metricRegistry = new MetricRegistry();
-
-        try (EventContextConsumer eventContextConsumer = new EventContextConsumer(configSource, new OutputFake(), metricRegistry, prometheusPort)) {
-            final double records = 10;
-            long length = 0L;
-            for (int i = 0; i < records; i++) {
-                EventContext eventContext = eventContextFactory.create();
-                length = length + eventContext.getEventData().getBody().length;
-                eventContextConsumer.accept(eventContext);
+    public void testDepthBytesMetric() {
+        EventContext eventContext = eventContextFactory.create();
+        eventContextConsumer.accept(eventContext);
+
+        long depth1 = 0L;
+        final double records = 10;
+        for (int i = 1; i < records; i++) { // records - 1 loops
+            if (i == 5) { // 5 records per partition
```

Comment:
is there a reason this was changed from i==4 to i==5?

Reply:
It still checks the fifth eventContext created. This is difficult to see in the GitHub review tool, but the if clause was moved from the end of the loop to the start of the loop. Before it created a new context and then checked the if statement, now it checks the if statement and then creates a new context, so that's why the index was incremented by one in the if statement.